### ISSUE TITLE
Not OK: perl v5.7.3 on cygwin 1.3.9(0.5132)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5204]
Migrated from [rt.perl.org#8774](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8774) (status was 'resolved')

Searchable as RT8774$

**Comment by p5pRT** [comment]
### From @JohnPeacock



Failed Test               Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\.\./ext/Encode/t/JP\.t         4  1024    22    4  18\.18%  9 14 18 21
\.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
op/groups\.t                              2    2 100\.00%  1\-2

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=none

Site configuration information for perl v5.7.3:

Configured by jpeacock at Tue Mar  5 09:38:25  2002.

Summary of my perl5 (revision 5.0 version 7 subversion 3) configuration:
  Platform:
    osname=cygwin, osvers=1.3.9(0.5132), archname=cygwin
    uname='cygwin_nt-5.0 jpeacock 1.3.9(0.5132) 2002-01-21 12:48 i686 unknown '
    config_args='-des -Dusedevel'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing',
    optimize='-O2',
    cppflags='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing'
    ccversion='', gccversion='2.95.3-5 (cygwin special)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ld2', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-lgdbm -lcrypt -lutil
    perllibs=-lcrypt -lutil
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.7.3:
    ../lib
    /usr/local/lib/perl5/5.7.3/cygwin
    /usr/local/lib/perl5/5.7.3
    /usr/local/lib/perl5/site_perl/5.7.3/cygwin
    /usr/local/lib/perl5/site_perl/5.7.3
    /usr/local/lib/perl5/site_perl/5.7.2/cygwin
    /usr/local/lib/perl5/site_perl/5.7.2
    /usr/local/lib/perl5/site_perl/5.6.1/cygwin
    /usr/local/lib/perl5/site_perl/5.6.1
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.7.3:
    CYGWIN=ntea ntsec error_start=d:/CygWin/bin/debug_wrapper.bat
    HOME=/home/jpeacock
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/X11R6/bin:/usr/local/bin:/usr/bin:/bin
    PERLDB_OPTS=
    PERL_BADLANG (unset)
    SHELL=bash



```
</details>

**Comment by p5pRT** [comment]
### From @tux

On Tue 05 Mar 2002 20&#8203;:18\, John Peacock \<jpeacock@&#8203;rowman\.com> wrote&#8203;:
> 
> This is a build failure report for perl from jpeacock@&#8203;rowman\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.7\.3\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Failed Test               Stat Wstat Total Fail  Failed  List of Failed
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \.\./ext/Encode/t/JP\.t         4  1024    22    4  18\.18%  9 14 18 21
> \.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
> op/groups\.t                              2    2 100\.00%  1\-2

Can you give me a full run of the groups\.t please?
What is your $CYGWIN setting? 'ntsec' is in there I hope?

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.7\.2 & 631 on HP\-UX 10\.20 & 11\.00\, AIX 4\.2\, AIX 4\.3\,
&nbsp;  WinNT 4\, Win2K pro & WinCE 2\.11\.  Smoking perl CORE&#8203;: smokers@&#8203;perl\.org
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From @schwern

On Tue\, Mar 05\, 2002 at 02&#8203;:18&#8203;:58PM \-0500\, John Peacock wrote&#8203;:
> \.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9

Known problem\.  I forget if I fixed it\.  Could you try&#8203;:
http&#8203;://www\.cpan\.org/authors/id/M/MS/MSCHWERN/ExtUtils\-MakeMaker\-5\.50\_01\.tar\.gz


\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
Carpe canem\!  Seize the dog\!  This cannot be right\.
&nbsp;    \-\- The Critic

**Comment by p5pRT** [comment]
### From @JohnPeacock

H\.Merijn Brand wrote&#8203;:
> Can you give me a full run of the groups\.t please?
> What is your $CYGWIN setting? 'ntsec' is in there I hope?
> 

Jarkko added this to the output for "perl \-V" \(at my urging\)\.\.\.

Environment for perl v5\.7\.3&#8203;:
&nbsp;     CYGWIN=ntea ntsec error\_start=d&#8203;:/CygWin/bin/debug\_wrapper\.bat

> 

$ \./perl op/groups\.t
\# groups = uid=1002\(jpeacock\) gid=513\(None\) 
groups=0\(Everyone\)\,544\(Administrators\)\,545\(Users\)\,0\(Everyone\)\,11472\(ClearCase\)\,10512\(Domain 
Admins\)\,10513\(Domain Users\)\,11607\(Webserver\)
\# 
groups=0\(Everyone\)\,544\(Administrators\)\,545\(Users\)\,0\(Everyone\)\,11472\(ClearCase\)\,10512\(Domain 
Admins\)\,10513\(Domain Users\)\,11607\(Webserver\)
\# g0 = 0\(Everyone\) 544\(Administrators\) 545\(Users\) 0\(Everyone\) 11472\(ClearCase\) 
10512\(Domain Admins\) 10513\(Domain Users\) 11607\(Webserver\)
\# g1 = Everyone Administrators Users Everyone ClearCase Domain Admins Domain 
Users Webserver
1\.\.2
\# pwgid = 513\, pwgnam = None
\# gr = Webserver Domain Users Domain Admins ClearCase Everyone Users Administrators
\#gr1 is \<Administrators ClearCase Domain Admins Domain Users Everyone Users 
Webserver>
\#gr2 is \<Administrators Admins ClearCase Domain Everyone Users Webserver>
not ok 1
not ok 2 \(groupstype should be type short\, not long\)

I'm sorry\, I haven't run the testing much recently \(trying to get other
stuff done\)\.\.\.

John

\-\- 
John Peacock
Director of Information Research and Technology
Rowman & Littlefield Publishing Group
4720 Boston Way
Lanham\, MD 20706
301\-459\-3366 x\.5010
fax 301\-429\-5747

**Comment by p5pRT** [comment]
### From @JohnPeacock

Michael G Schwern wrote&#8203;:
> On Tue\, Mar 05\, 2002 at 02&#8203;:18&#8203;:58PM \-0500\, John Peacock wrote&#8203;:
> 
>>\.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
>>
> 
> Known problem\.  I forget if I fixed it\.  Could you try&#8203;:
> http&#8203;://www\.cpan\.org/authors/id/M/MS/MSCHWERN/ExtUtils\-MakeMaker\-5\.50\_01\.tar\.gz
> 

Do I just copy the files into lib or do I need to do something more
creative?

Just FYI\, I see this when I run that test manually&#8203;:

$ \./perl \-I\.\./lib \-I\.\. \.\./lib/ExtUtils/t/Embed\.t
1\.\.9
cp&#8203;: cannot stat \`\.\./cygperl5\_7\_3\.dll'&#8203;: No such file or directory
\# gcc \-o embed\_test\.exe \-I\.\.  \-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing 
\-I/usr/local/lib/perl5/5\.7\.3/cygwin/CORE  embed\_test\.c \-L\.\. \-lperl 
\-L/usr/local/lib \.\./lib/auto/DynaLoader/DynaLoader\.a \-lcrypt \-lutil
ok 1
\# embed\_test = \./embed\_test\.exe
not ok 9 \# 32768

then\, I get a requester that complains "The dynamic link library 
cygperl5\_7\_2\.dll could not be found in the specified path\."  I suspect
there is significance to the different versions mentioned\.

John


\-\- 
John Peacock
Director of Information Research and Technology
Rowman & Littlefield Publishing Group
4720 Boston Way
Lanham\, MD 20706
301\-459\-3366 x\.5010
fax 301\-429\-5747

**Comment by p5pRT** [comment]
### From @schwern

On Tue\, Mar 05\, 2002 at 03&#8203;:24&#8203;:30PM \-0500\, John Peacock wrote&#8203;:
> Michael G Schwern wrote&#8203;:
> >On Tue\, Mar 05\, 2002 at 02&#8203;:18&#8203;:58PM \-0500\, John Peacock wrote&#8203;:
> >
> >>\.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
> >>
> >
> >Known problem\.  I forget if I fixed it\.  Could you try&#8203;:
> >http&#8203;://www\.cpan\.org/authors/id/M/MS/MSCHWERN/ExtUtils\-MakeMaker\-5\.50\_01\.tar\.gz
> >
> 
> Do I just copy the files into lib or do I need to do something more
> creative?

Just test it like any other CPAN module with any ol' perl 5\.6\.1\.
Doesn't have to be on 5\.7\.3\.  The problem is in the
test/ExtUtils&#8203;::Embed constructing the proper gcc incantations as
opposed to anything in perl itself \(as you've already seen below\)\.


> Just FYI\, I see this when I run that test manually&#8203;:
> 
> $ \./perl \-I\.\./lib \-I\.\. \.\./lib/ExtUtils/t/Embed\.t
> 1\.\.9
> cp&#8203;: cannot stat \`\.\./cygperl5\_7\_3\.dll'&#8203;: No such file or directory
> \# gcc \-o embed\_test\.exe \-I\.\.  \-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing 
> \-I/usr/local/lib/perl5/5\.7\.3/cygwin/CORE  embed\_test\.c \-L\.\. \-lperl 
> \-L/usr/local/lib \.\./lib/auto/DynaLoader/DynaLoader\.a \-lcrypt \-lutil
> ok 1
> \# embed\_test = \./embed\_test\.exe
> not ok 9 \# 32768
> 
> then\, I get a requester that complains "The dynamic link library 
> cygperl5\_7\_2\.dll could not be found in the specified path\."  I suspect
> there is significance to the different versions mentioned\.




\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
my anus yearns now
warm paste fills me happily
saturday morning
&nbsp;    \-\- imploded

**Comment by p5pRT** [comment]
### From @JohnPeacock

Michael G Schwern wrote&#8203;:
> 
> Just test it like any other CPAN module with any ol' perl 5\.6\.1\.
> Doesn't have to be on 5\.7\.3\.  The problem is in the
> test/ExtUtils&#8203;::Embed constructing the proper gcc incantations as
> opposed to anything in perl itself \(as you've already seen below\)\.

$ make
cp lib/ExtUtils/MM\_NW5\.pm blib/lib/ExtUtils/MM\_NW5\.pm
cp lib/ExtUtils/Liblist\.pm blib/lib/ExtUtils/Liblist\.pm
cp lib/ExtUtils/MM\_Cygwin\.pm blib/lib/ExtUtils/MM\_Cygwin\.pm
cp lib/ExtUtils/testlib\.pm blib/lib/ExtUtils/testlib\.pm
cp lib/ExtUtils/MM\_OS2\.pm blib/lib/ExtUtils/MM\_OS2\.pm
cp lib/ExtUtils/MM\_Win95\.pm blib/lib/ExtUtils/MM\_Win95\.pm
cp lib/ExtUtils/Installed\.pm blib/lib/ExtUtils/Installed\.pm
cp lib/ExtUtils/Command/MM\.pm blib/lib/ExtUtils/Command/MM\.pm
cp lib/ExtUtils/typemap blib/lib/ExtUtils/typemap
cp lib/ExtUtils/Mksymlists\.pm blib/lib/ExtUtils/Mksymlists\.pm
cp lib/ExtUtils/Packlist\.pm blib/lib/ExtUtils/Packlist\.pm
cp lib/ExtUtils/MM\_Unix\.pm blib/lib/ExtUtils/MM\_Unix\.pm
Can't locate ExtUtils/MM\_Any\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: blib/arch blib/lib 
/usr/local/lib/perl5/5\.6\.1/cygwin /usr/local/lib/perl5/5\.6\.1 
/usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin /usr/local/lib/perl5/site\_perl/5\.6\.1 
/usr/local/lib/perl5/site\_perl \.\) at blib/lib/ExtUtils/MM\_Unix\.pm line 21\.
Compilation failed in require at 
/usr/local/lib/perl5/5\.6\.1/ExtUtils/MakeMaker\.pm line 75\.

Hmmm\.\.\.  with 5\.7\.2 it gets farther and then complains about ExtUtils/MY\.pm
instead \(even though it is there\)&#8203;:

$ make
cp lib/ExtUtils/Installed\.pm blib/lib/ExtUtils/Installed\.pm
cp lib/ExtUtils/MM\_OS2\.pm blib/lib/ExtUtils/MM\_OS2\.pm
cp lib/ExtUtils/MakeMaker\.pm blib/lib/ExtUtils/MakeMaker\.pm
cp lib/ExtUtils/Install\.pm blib/lib/ExtUtils/Install\.pm
cp lib/ExtUtils/typemap blib/lib/ExtUtils/typemap
cp lib/ExtUtils/Command\.pm blib/lib/ExtUtils/Command\.pm
cp lib/ExtUtils/Manifest\.pm blib/lib/ExtUtils/Manifest\.pm
cp lib/ExtUtils/MM\_Unix\.pm blib/lib/ExtUtils/MM\_Unix\.pm
cp lib/ExtUtils/Mksymlists\.pm blib/lib/ExtUtils/Mksymlists\.pm
cp lib/ExtUtils/MM\.pm blib/lib/ExtUtils/MM\.pm
cp lib/ExtUtils/Embed\.pm blib/lib/ExtUtils/Embed\.pm
cp lib/ExtUtils/MANIFEST\.SKIP blib/lib/ExtUtils/MANIFEST\.SKIP
cp lib/ExtUtils/MM\_UWIN\.pm blib/lib/ExtUtils/MM\_UWIN\.pm
cp lib/ExtUtils/MM\_Win32\.pm blib/lib/ExtUtils/MM\_Win32\.pm
cp lib/ExtUtils/testlib\.pm blib/lib/ExtUtils/testlib\.pm
cp lib/ExtUtils/MM\_Cygwin\.pm blib/lib/ExtUtils/MM\_Cygwin\.pm
cp lib/ExtUtils/MM\_Win95\.pm blib/lib/ExtUtils/MM\_Win95\.pm
cp lib/ExtUtils/MM\_VMS\.pm blib/lib/ExtUtils/MM\_VMS\.pm
cp lib/ExtUtils/MY\.pm blib/lib/ExtUtils/MY\.pm
cp lib/ExtUtils/MM\_DOS\.pm blib/lib/ExtUtils/MM\_DOS\.pm
cp lib/ExtUtils/MM\_BeOS\.pm blib/lib/ExtUtils/MM\_BeOS\.pm
cp lib/ExtUtils/Liblist\.pm blib/lib/ExtUtils/Liblist\.pm
cp lib/ExtUtils/Packlist\.pm blib/lib/ExtUtils/Packlist\.pm
cp lib/ExtUtils/Liblist/Kid\.pm blib/lib/ExtUtils/Liblist/Kid\.pm
cp lib/ExtUtils/Mkbootstrap\.pm blib/lib/ExtUtils/Mkbootstrap\.pm
cp lib/ExtUtils/Constant\.pm blib/lib/ExtUtils/Constant\.pm
cp lib/ExtUtils/Command/MM\.pm blib/lib/ExtUtils/Command/MM\.pm
cp lib/ExtUtils/MM\_NW5\.pm blib/lib/ExtUtils/MM\_NW5\.pm
cp lib/ExtUtils/MM\_Any\.pm blib/lib/ExtUtils/MM\_Any\.pm
cp bin/xsubpp blib/script/xsubpp
/usr/local/bin/perl5\.7\.2\.exe \-MExtUtils&#8203;::MY \-e "MY\->fixin\(shift\)" blib/script/xsubpp
Can't locate ExtUtils/MY\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: 
/usr/local/lib/perl5/5\.7\.2/cygwin /usr/local/lib/perl5/5\.7\.2 
/usr/local/lib/perl5/site\_perl/5\.7\.2/cygwin /usr/local/lib/perl5/site\_perl/5\.7\.2 
/usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin /usr/local/lib/perl5/site\_perl/5\.6\.1 
/usr/local/lib/perl5/site\_perl \.\)\.
BEGIN failed\-\-compilation aborted\.
make&#8203;: \*\*\* \[blib/script/xsubpp\] Error 2

But this worked&#8203;:

$ perl5\.7\.2 \-Iblib/lib \-MExtUtils&#8203;::MY \-e "MY\->fixin\(shift\)" blib/script/xsubpp

and they trying 'make' twice in a row eventually got me past that\.  However\, I
still get&#8203;:

t/Embed\.\.\.\.\.\.\.\.\.\.NOK 1Can't exec "\./embed\_test\.exe"&#8203;: No such file or directory 
at t/Embed\.t line 153\.
t/Embed\.\.\.\.\.\.\.\.\.\.NOK 9Test output counter mismatch \[test 9\]
t/Embed\.\.\.\.\.\.\.\.\.\.FAILED tests 1\-9
&nbsp;         Failed 9/9 tests\, 0\.00% okay

Running Embed\.t from the \./t directory itself\, I get&#8203;:

$ perl \-I\.\./blib/lib Embed\.t
1\.\.9
\# gcc \-o embed\_test\.exe \-I\.\.  \-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing 
\-I/usr/local/lib/perl5/5\.6\.1/cygwin/CORE  embed\_test\.c \-L\.\. \-lperl  \-s  \-s 
\-L/usr/local/lib /usr/local/lib/perl5/5\.6\.1/cygwin/auto/DynaLoader/DynaLoader\.a 
\-L/usr/local/lib/perl5/5\.6\.1/cygwin/CORE \-lcrypt
/usr/lib/gcc\-lib/i686\-pc\-cygwin/2\.95\.3\-5/\.\./\.\./\.\./\.\./i686\-pc\-cygwin/bin/ld&#8203;: 
cannot find \-lperl
collect2&#8203;: ld returned 1 exit status
not ok 1
\# embed\_test = \./embed\_test\.exe
not ok 9 \# \-1


John

\-\- 
John Peacock
Director of Information Research and Technology
Rowman & Littlefield Publishing Group
4720 Boston Way
Lanham\, MD 20706
301\-459\-3366 x\.5010
fax 301\-429\-5747

**Comment by p5pRT** [comment]
### From @schwern

On Tue\, Mar 05\, 2002 at 03&#8203;:57&#8203;:38PM \-0500\, John Peacock wrote&#8203;:
> Can't locate ExtUtils/MM\_Any\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: blib/arch blib/lib 
> /usr/local/lib/perl5/5\.6\.1/cygwin /usr/local/lib/perl5/5\.6\.1 
> /usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin 
> /usr/local/lib/perl5/site\_perl/5\.6\.1 /usr/local/lib/perl5/site\_perl \.\) at 
> blib/lib/ExtUtils/MM\_Unix\.pm line 21\.
> Compilation failed in require at 
> /usr/local/lib/perl5/5\.6\.1/ExtUtils/MakeMaker\.pm line 75\.

Ok\, I've replicated this\.


> /usr/local/bin/perl5\.7\.2\.exe \-MExtUtils&#8203;::MY \-e "MY\->fixin\(shift\)" 
> blib/script/xsubpp
> Can't locate ExtUtils/MY\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: 
> /usr/local/lib/perl5/5\.7\.2/cygwin /usr/local/lib/perl5/5\.7\.2 
> /usr/local/lib/perl5/site\_perl/5\.7\.2/cygwin 
> /usr/local/lib/perl5/site\_perl/5\.7\.2 
> /usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin 
> /usr/local/lib/perl5/site\_perl/5\.6\.1 /usr/local/lib/perl5/site\_perl \.\)\.
> BEGIN failed\-\-compilation aborted\.
> make&#8203;: \*\*\* \[blib/script/xsubpp\] Error 2
> 
> But this worked&#8203;:
> 
> $ perl5\.7\.2 \-Iblib/lib \-MExtUtils&#8203;::MY \-e "MY\->fixin\(shift\)" 
> blib/script/xsubpp

Ok\, that's an easy tweak\.

The basic problem is MakeMaker has to use itself to install itself\,
but its not being consistent about making sure all the \-I's are set\.

I'll have to replace a good chunk of the $\(PERLRUN\)'s with
$\(PERLRUNINST\) so it looks in blib for MakeMaker\.


But Embed\.t is still broken\. :\(  I'll poke at it\.


\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
The eye opening delightful morning taste of expired cheese bits in sour milk\!

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

John Peacock \<jpeacock@&#8203;rowman\.com> writes&#8203;:
>This is a build failure report for perl from jpeacock@&#8203;rowman\.com\,
>generated with the help of perlbug 1\.33 running under perl v5\.7\.3\.
>
>
>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>\[Please enter your report here\]
>
>Failed Test               Stat Wstat Total Fail  Failed  List of Failed
>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>\.\./ext/Encode/t/JP\.t         4  1024    22    4  18\.18%  9 14 18 21

I really meant to fix that one before we shipped \- but could not get
gremlins out of my machine in time\.

Can someone/you convert ext/Encode/t/table\.euc and ext/Encode/t/table\.ref
to CRLF line endings and try again? \- it should then pass\.\.\.

\-\-
Nick Ing\-Simmons
http&#8203;://www\.ni\-s\.u\-net\.com/

**Comment by p5pRT** [comment]
### From @tux

On Tue 05 Mar 2002 21&#8203;:08\, John Peacock \<jpeacock@&#8203;rowman\.com> wrote&#8203;:
> H\.Merijn Brand wrote&#8203;:
> > Can you give me a full run of the groups\.t please?
> > What is your $CYGWIN setting? 'ntsec' is in there I hope?
> > 
> 
> Jarkko added this to the output for "perl \-V" \(at my urging\)\.\.\.
> 
> Environment for perl v5\.7\.3&#8203;:
>      CYGWIN=ntea ntsec error\_start=d&#8203;:/CygWin/bin/debug\_wrapper\.bat

I seriously doubt the usefulness of this info in \-V\.

How should we at p5p interpret this? Is it the setting that was used when perl
was built? Or was it the setting the user had active when he ran the test? Or
was it the setting when he reported the bug? All could be different you know
and all would have a different impact\.

In your case I presume and hope they all are the same :\)

I'll look into this problem\, but not on a very short notice \.\.\.
Remind me if I keep silent for more than two weeks\.

> $ \./perl op/groups\.t
> \# groups = uid=1002\(jpeacock\) gid=513\(None\) 
> groups=0\(Everyone\)\,544\(Administrators\)\,545\(Users\)\,0\(Everyone\)\,11472\(ClearCase\)\,10512\(Domain 
> Admins\)\,10513\(Domain Users\)\,11607\(Webserver\)
> \# 
> groups=0\(Everyone\)\,544\(Administrators\)\,545\(Users\)\,0\(Everyone\)\,11472\(ClearCase\)\,10512\(Domain 
> Admins\)\,10513\(Domain Users\)\,11607\(Webserver\)
> \# g0 = 0\(Everyone\) 544\(Administrators\) 545\(Users\) 0\(Everyone\) 11472\(ClearCase\) 
> 10512\(Domain Admins\) 10513\(Domain Users\) 11607\(Webserver\)
> \# g1 = Everyone Administrators Users Everyone ClearCase Domain Admins Domain 
> Users Webserver
> 1\.\.2
> \# pwgid = 513\, pwgnam = None
> \# gr = Webserver Domain Users Domain Admins ClearCase Everyone Users Administrators
> \#gr1 is \<Administrators ClearCase Domain Admins Domain Users Everyone Users 
> Webserver>
> \#gr2 is \<Administrators Admins ClearCase Domain Everyone Users Webserver>
> not ok 1
> not ok 2 \(groupstype should be type short\, not long\)
> 
> I'm sorry\, I haven't run the testing much recently \(trying to get other
> stuff done\)\.\.\.

Thanks for the feedback

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.7\.2 & 631 on HP\-UX 10\.20 & 11\.00\, AIX 4\.2\, AIX 4\.3\,
&nbsp;  WinNT 4\, Win2K pro & WinCE 2\.11\.  Smoking perl CORE&#8203;: smokers@&#8203;perl\.org
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From @JohnPeacock

H\.Merijn Brand > On Tue 05 Mar 2002 21&#8203;:08\, John Peacock \<jpeacock@&#8203;rowman\.com> wrote&#8203;:
>>Jarkko added this to the output for "perl \-V" \(at my urging\)\.\.\.
>>
>>Environment for perl v5\.7\.3&#8203;:
>>     CYGWIN=ntea ntsec error\_start=d&#8203;:/CygWin/bin/debug\_wrapper\.bat
>>
> 
> I seriously doubt the usefulness of this info in \-V\.
> 
> How should we at p5p interpret this? Is it the setting that was used when perl
> was built? Or was it the setting the user had active when he ran the test? Or
> was it the setting when he reported the bug? All could be different you know
> and all would have a different impact\.

All of the environment variables listed by \-V are the active environment\, not
something squirreled away in %CONFIG\, which is as it should be\.  I just checked
to be sure\.  I would argue that this \*is\* useful information\, since anyone on
p5p who is trying to respond to a bug report on Cygwin can quickly check for
known failure modes \(like missing 'ntsec'\)\.  We do have to make the \[reasonable\]
assumption that the perlbug was run in the same environment as the bug itself\.

John

\-\- 
John Peacock
Director of Information Research and Technology
Rowman & Littlefield Publishing Group
4720 Boston Way
Lanham\, MD 20706
301\-459\-3366 x\.5010
fax 301\-429\-5747

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Hallo John\,

Am 2002\-03\-05 um 21&#8203;:24 schriebst du&#8203;:

> Michael G Schwern wrote&#8203;:
>> On Tue\, Mar 05\, 2002 at 02&#8203;:18&#8203;:58PM \-0500\, John Peacock wrote&#8203;:
>> 
>>>\.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
>>>
>> 
>> Known problem\.  I forget if I fixed it\.  Could you try&#8203;:
>> http&#8203;://www\.cpan\.org/authors/id/M/MS/MSCHWERN/ExtUtils\-MakeMaker\-5\.50\_01\.tar\.gz
>> 

> Do I just copy the files into lib or do I need to do something more
> creative?

> Just FYI\, I see this when I run that test manually&#8203;:

> $ \./perl \-I\.\./lib \-I\.\. \.\./lib/ExtUtils/t/Embed\.t
> 1\.\.9
> cp&#8203;: cannot stat \`\.\./cygperl5\_7\_3\.dll'&#8203;: No such file or directory
> \# gcc \-o embed\_test\.exe \-I\.\.  \-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing 
> \-I/usr/local/lib/perl5/5\.7\.3/cygwin/CORE  embed\_test\.c \-L\.\. \-lperl 
> \-L/usr/local/lib \.\./lib/auto/DynaLoader/DynaLoader\.a \-lcrypt \-lutil
> ok 1
> \# embed\_test = \./embed\_test\.exe
> not ok 9 \# 32768

> then\, I get a requester that complains "The dynamic link library 
> cygperl5\_7\_2\.dll could not be found in the specified path\."  I suspect
> there is significance to the different versions mentioned\.

Hmmm\, probably some hardcoded filenames somewhere\, I'll take a look today\.


Gerrit
\-\- 
=^\.\.^=

**Comment by p5pRT** [comment]
### From @tux

On Fri 08 Mar 2002 09&#8203;:35\, "Gerrit P\. Haase" \<gp@&#8203;familiehaase\.de> wrote&#8203;:
> Hallo John\,
> 
> Am 2002\-03\-05 um 21&#8203;:24 schriebst du&#8203;:
> 
> > Do I just copy the files into lib or do I need to do something more
> > creative?
> 
> > Just FYI\, I see this when I run that test manually&#8203;:
> 
> > $ \./perl \-I\.\./lib \-I\.\. \.\./lib/ExtUtils/t/Embed\.t
> > 1\.\.9
> > cp&#8203;: cannot stat \`\.\./cygperl5\_7\_3\.dll'&#8203;: No such file or directory
> > \# gcc \-o embed\_test\.exe \-I\.\.  \-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing 
> > \-I/usr/local/lib/perl5/5\.7\.3/cygwin/CORE  embed\_test\.c \-L\.\. \-lperl 
> > \-L/usr/local/lib \.\./lib/auto/DynaLoader/DynaLoader\.a \-lcrypt \-lutil
> > ok 1
> > \# embed\_test = \./embed\_test\.exe
> > not ok 9 \# 32768
> 
> > then\, I get a requester that complains "The dynamic link library 
> > cygperl5\_7\_2\.dll could not be found in the specified path\."  I suspect
> > there is significance to the different versions mentioned\.
> 
> Hmmm\, probably some hardcoded filenames somewhere\, I'll take a look today\.

Already done\. \# 15062

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.7\.2 & 631 on HP\-UX 10\.20 & 11\.00\, AIX 4\.2\, AIX 4\.3\,
&nbsp;  WinNT 4\, Win2K pro & WinCE 2\.11\.  Smoking perl CORE&#8203;: smokers@&#8203;perl\.org
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

> John Peacock \<jpeacock@&#8203;rowman\.com> writes&#8203;:
> >This is a build failure report for perl from jpeacock@&#8203;rowman\.com\,
> >generated with the help of perlbug 1\.33 running under perl v5\.7\.3\.
> >
> >
> >\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> >\[Please enter your report here\]
> >
> >Failed Test               Stat Wstat Total Fail  Failed  List of Failed
> >\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> >\.\./ext/Encode/t/JP\.t         4  1024    22    4  18\.18%  9 14 18 21
> 
> I really meant to fix that one before we shipped \- but could not get
> gremlins out of my machine in time\.
> 
> Can someone/you convert ext/Encode/t/table\.euc and ext/Encode/t/table\.ref
> to CRLF line endings and try again? \- it should then pass\.\.\.
Yes\, it does \( Win2k \)

Regards
Mattia

**Comment by p5pRT** [comment]
### From @jhi

I tried 15205 and was one test luckier&#8203;:

Failed Test          Stat Wstat Total Fail  Failed  List of Failed


---
\.\.\./ext/Encode/t/JP\.t    4  1024    22    4  18\.18%  9 14 18 21
io/fs\.t                            36    4  11\.11%  5 7 9\-10
44 tests and 372 subtests skipped\.
Failed 2/619 test scripts\, 99\.68% okay\. 8/55498 subtests failed\, 99\.99% okay\.

$ \./myconfig
Summary of my perl5 \(revision 5\.0 version 7 subversion 3 patch 15172\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=cygwin\, osvers=1\.3\.10\(0\.5132\)\, archname=cygwin
&nbsp;    uname='cygwin\_nt\-5\.0 foobar 1\.3\.10\(0\.5132\) 2002\-02\-25 11&#8203;:14 i686 unknown '
&nbsp;    config\_args='\-des \-Dusedevel'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=y\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, ccflags ='\-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing'\,
&nbsp;    optimize='\-O2'\,
&nbsp;    cppflags='\-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing'
&nbsp;    ccversion=''\, gccversion='2\.95\.3\-5 \(cygwin special\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=4
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld2'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib /lib
&nbsp;    libs=\-lgdbm \-lcrypt \-lutil
&nbsp;    perllibs=\-lcrypt \-lutil
&nbsp;    libc=/usr/lib/libc\.a\, so=dll\, useshrplib=true\, libperl=libperl
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-L/usr/local/lib'

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @schwern

On Tue\, Mar 05\, 2002 at 03&#8203;:57&#8203;:38PM \-0500\, John Peacock wrote&#8203;:
> Can't locate ExtUtils/MM\_Any\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: blib/arch blib/lib 
> /usr/local/lib/perl5/5\.6\.1/cygwin /usr/local/lib/perl5/5\.6\.1 
> /usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin 
> /usr/local/lib/perl5/site\_perl/5\.6\.1 /usr/local/lib/perl5/site\_perl \.\) at 
> blib/lib/ExtUtils/MM\_Unix\.pm line 21\.
> Compilation failed in require at 
> /usr/local/lib/perl5/5\.6\.1/ExtUtils/MakeMaker\.pm line 75\.

Ok\, I've fixed this\.  Could you try pulling a copy of MakeMaker out of
the CVS repository and try it out?  Treat it like a CPAN module\.
http&#8203;://archive\.develooper\.com/makemaker@&#8203;perl\.org/msg00214\.html


\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
Try explaining that to my brain which is currently ON FIRE\!
&nbsp;    http&#8203;://www\.goats\.com/archive/010704\.html

**Comment by p5pRT** [comment]
### From @JohnPeacock

Michael G Schwern wrote&#8203;:
> On Tue\, Mar 05\, 2002 at 03&#8203;:57&#8203;:38PM \-0500\, John Peacock wrote&#8203;:
> 
>>Can't locate ExtUtils/MM\_Any\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: blib/arch blib/lib 
>>/usr/local/lib/perl5/5\.6\.1/cygwin /usr/local/lib/perl5/5\.6\.1 
>>/usr/local/lib/perl5/site\_perl/5\.6\.1/cygwin 
>>/usr/local/lib/perl5/site\_perl/5\.6\.1 /usr/local/lib/perl5/site\_perl \.\) at 
>>blib/lib/ExtUtils/MM\_Unix\.pm line 21\.
>>Compilation failed in require at 
>>/usr/local/lib/perl5/5\.6\.1/ExtUtils/MakeMaker\.pm line 75\.
> 
> 
> Ok\, I've fixed this\.  Could you try pulling a copy of MakeMaker out of
> the CVS repository and try it out?  Treat it like a CPAN module\.
> http&#8203;://archive\.develooper\.com/makemaker@&#8203;perl\.org/msg00214\.html
> 

OK\, that fixes the above problem\, but of course I still get the actual test 
failures under 5\.6\.1&#8203;:

PERL\_DL\_NONLAZY=1 /usr/local/bin/perl\.exe "\-I/d/working/ExtUtils\-MakeMaker/lib" 
"\-I/d/working/ExtUtils\-MakeMaker/blib/arch" 
"\-I/d/working/ExtUtils\-MakeMaker/blib/lib" "\-MExtUtils&#8203;::Command&#8203;::MM" "\-e" 
"test\_harness\(0\)" t/\*\.t
t/Command\.\.\.\.\.\.\.\.\.\.\.ok\, 3/24 skipped&#8203;: different file permission semantics on cygwin
t/Embed\.\.\.\.\.\.\.\.\.\.\.\.\./usr/lib/gcc\-lib/i686\-pc\-cygwin/2\.95\.3\-5/\.\./\.\./\.\./\.\./i686\-pc\-cygwin/bin/ld&#8203;: 
cannot find \-lperl
collect2&#8203;: ld returned 1 exit status
Can't spawn "\./embed\_test\.exe"&#8203;: No such file or directory at t/Embed\.t line 151\.
t/Embed\.\.\.\.\.\.\.\.\.\.\.\.\.FAILED tests 1\-9
&nbsp;         Failed 9/9 tests\, 0\.00% okay
t/ExtUtils\.\.\.\.\.\.\.\.\.\.ok 2/27make\[1\]&#8203;: \*\*\* No rule to make target 
\`/usr/local/lib/perl5/5\.6\.1/cygwin/CORE/libperl5\_6\_1\.dll\.a'\, needed by 
\`blib/arch/auto/ExtTest/ExtTest\.dll'\.  Stop\.
t/ExtUtils\.\.\.\.\.\.\.\.\.\.FAILED tests 3\-27
&nbsp;         Failed 25/27 tests\, 7\.41% okay
t/Installed\.\.\.\.\.\.\.\.\.ok
t/MM\_BeOS\.\.\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not BeOS
t/MM\_Cygwin\.\.\.\.\.\.\.\.\.Name "MM&#8203;::perl\_script" used only once&#8203;: possible typo at 
t/MM\_Cygwin\.t line 78\.
Name "ExtUtils&#8203;::MM\_Unix&#8203;::cflags" used only once&#8203;: possible typo at t/MM\_Cygwin\.t 
line 48\.
t/MM\_Cygwin\.\.\.\.\.\.\.\.\.ok\, 2/15 skipped&#8203;: Only relevent in the core
t/MM\_OS2\.\.\.\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not OS/2
t/MM\_Unix\.\.\.\.\.\.\.\.\.\.\.skipped&#8203;: Non\-Unix platform
t/MM\_VMS\.\.\.\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not VMS
t/MM\_Win32\.\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not Win32
t/Manifest\.\.\.\.\.\.\.\.\.\.ok
t/Mkbootstrap\.\.\.\.\.\.\.ok
t/Packlist\.\.\.\.\.\.\.\.\.\.ok
t/basic\.\.\.\.\.\.\.\.\.\.\.\.\.ok
t/hints\.\.\.\.\.\.\.\.\.\.\.\.\.ok
t/testlib\.\.\.\.\.\.\.\.\.\.\.ok
Failed Test  Status Wstat Total Fail  Failed  List of Failed


---
t/Embed\.t                      9    9 100\.00%  1\-9
t/ExtUtils\.t                  27   25  92\.59%  3\-27
5 tests and 5 subtests skipped\.


and here under 5\.7\.3 \(DEVEL15108\)&#8203;:

PERL\_DL\_NONLAZY=1 /usr/local/bin/perl5\.7\.3\.exe 
"\-I/d/working/ExtUtils\-MakeMaker/lib" 
"\-I/d/working/ExtUtils\-MakeMaker/blib/arch" 
"\-I/d/working/ExtUtils\-MakeMaker/blib/lib" "\-MExtUtils&#8203;::Command&#8203;::MM" "\-e" 
"test\_harness\(0\)" t/\*\.t
t/Command\.\.\.\.\.\.\.\.ok\, 3/24 skipped&#8203;: different file permission semantics on cygwin
t/Embed\.\.\.\.\.\.\.\.\.\.ok
t/ExtUtils\.\.\.\.\.\.\.ok 2/27Creating library file&#8203;: libExtTest\.dll\.a
t/ExtUtils\.\.\.\.\.\.\.ok
t/Installed\.\.\.\.\.\.ok
t/MM\_BeOS\.\.\.\.\.\.\.\.skipped&#8203;: This is not BeOS
t/MM\_Cygwin\.\.\.\.\.\.Name "MM&#8203;::perl\_script" used only once&#8203;: possible typo at 
t/MM\_Cygwin\.t line 78\.
Name "ExtUtils&#8203;::MM\_Unix&#8203;::cflags" used only once&#8203;: possible typo at t/MM\_Cygwin\.t 
line 48\.
t/MM\_Cygwin\.\.\.\.\.\.ok\, 2/15 skipped&#8203;: Only relevent in the core
t/MM\_OS2\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not OS/2
t/MM\_Unix\.\.\.\.\.\.\.\.skipped&#8203;: Non\-Unix platform
t/MM\_VMS\.\.\.\.\.\.\.\.\.skipped&#8203;: This is not VMS
t/MM\_Win32\.\.\.\.\.\.\.skipped&#8203;: This is not Win32
t/Manifest\.\.\.\.\.\.\.ok
t/Mkbootstrap\.\.\.\.ok
t/Packlist\.\.\.\.\.\.\.ok
t/basic\.\.\.\.\.\.\.\.\.\.ok
t/hints\.\.\.\.\.\.\.\.\.\.ok
t/testlib\.\.\.\.\.\.\.\.ok
All tests successful\, 5 tests and 5 subtests skipped\.
Files=16\, Tests=217\, 101 wallclock secs \(17\.84 cusr \+ 10\.31 csys = 28\.15 CPU\)

Huzzah\!

John

\-\- 
John Peacock
Director of Information Research and Technology
Rowman & Littlefield Publishing Group
4720 Boston Way
Lanham\, MD 20706
301\-459\-3366 x\.5010
fax 301\-429\-5747

**Comment by p5pRT** [comment]
### From @JohnPeacock



Failed Test               Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\.\./ext/Encode/t/JP\.t         4  1024    22    4  18\.18%  9 14 18 21
\.\./lib/ExtUtils/t/Embed\.t                9    8  88\.89%  2\-9
op/groups\.t                              2    2 100\.00%  1\-2

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=none

Site configuration information for perl v5.7.3:

Configured by jpeacock at Tue Mar  5 09:38:25  2002.

Summary of my perl5 (revision 5.0 version 7 subversion 3) configuration:
  Platform:
    osname=cygwin, osvers=1.3.9(0.5132), archname=cygwin
    uname='cygwin_nt-5.0 jpeacock 1.3.9(0.5132) 2002-01-21 12:48 i686 unknown '
    config_args='-des -Dusedevel'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing',
    optimize='-O2',
    cppflags='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing'
    ccversion='', gccversion='2.95.3-5 (cygwin special)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ld2', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-lgdbm -lcrypt -lutil
    perllibs=-lcrypt -lutil
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.7.3:
    ../lib
    /usr/local/lib/perl5/5.7.3/cygwin
    /usr/local/lib/perl5/5.7.3
    /usr/local/lib/perl5/site_perl/5.7.3/cygwin
    /usr/local/lib/perl5/site_perl/5.7.3
    /usr/local/lib/perl5/site_perl/5.7.2/cygwin
    /usr/local/lib/perl5/site_perl/5.7.2
    /usr/local/lib/perl5/site_perl/5.6.1/cygwin
    /usr/local/lib/perl5/site_perl/5.6.1
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.7.3:
    CYGWIN=ntea ntsec error_start=d:/CygWin/bin/debug_wrapper.bat
    HOME=/home/jpeacock
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/X11R6/bin:/usr/local/bin:/usr/bin:/bin
    PERLDB_OPTS=
    PERL_BADLANG (unset)
    SHELL=bash


```
</details>
